src/dswx_sar/metadata.py

```diff
@@ -0,0 +1,311 @@
+
+    metadata_rtc_fields = {
+        'orbitPassDirection': 'str',
+        'burstID': 'str',
+        'productVersion': 'str',
+        'zeroDopplerStartTime': 'str',
+        'trackNumber': 'int',
+        'absoluteOrbitNumber': 'int'
+    }
+
+    # Mapping from RTC metadata to DSWx metadata
+    dswx_meta_mapping = {
+        'orbitPassDirection': 'ORBIT_PASS_DIRECTION',
+        'burstID': 'BURST_ID',
+        'productVersion': 'RTC_PRODUCT_VERSION',
+        'zeroDopplerStartTime': 'SENSING_TIME',
+        'trackNumber': 'TRACK_NUMBER',
+        'absoluteOrbitNumber': 'ABSOLUTE_ORBIT_NUMBER'
+    }
+
+    for h5_meta_path in h5path_list:
+        with h5py.File(h5_meta_path) as src:
+            for field, dtype in metadata_rtc_fields.items():
+                if dtype == 'str':
+                    value = str(src[f'/identification/{field}'].asstr()[...])
+
+                else:
+                    value = int(src[f'/identification/{field}'][...])
+                metadata_dict[field].append(value)
+
+    for rtc_field, dswx_field in dswx_meta_mapping.items():
+        values = metadata_dict[rtc_field]
+        
+        if dswx_field == 'SENSING_TIME':
+            start, end = _get_date_range(values)
+            dswx_metadata_dict['SENSING_START'] = start
+            dswx_metadata_dict['SENSING_END'] = end
+        else:
+            dswx_metadata_dict[dswx_field] = values[0] if len(set(values)) == 1 else values
+
+
+def _get_date_range(dates):
+    """Converts and returns the min and max date from a list of date strings."""
+    input_date_format = "%Y-%m-%dT%H:%M:%S"
+    output_date_format = "%Y-%m-%dT%H:%M:%SZ"
+    date_objects = [datetime.strptime(date[:19], input_date_format) for date in dates]
+    return datetime.strftime(min(date_objects), output_date_format), \
+           datetime.strftime(max(date_objects), output_date_format)
+
+
+def _populate_ancillary_metadata_datasets(dswx_metadata_dict, ancillary_cfg):
+    """Populate metadata dictionary with input files.
+    
+    Parameters
+    ----------
+    dswx_metadata_dict : collections.OrderedDict
+        Metadata dictionary.
+    ancillary_cfg: obj
+        Configuration object containing all ancillary data sources and their descriptions.
+    """
+
+    # Dictionary mapping of source type to its file and description attributes in ancillary_cfg
+    source_map = {
+        'DEM_SOURCE': ('dem_file', 'dem_file_description'),
+        'HAND_SOURCE': ('hand_file', 'hand_file_description'),
+        'WORLDCOVER_SOURCE': ('worldcover_file', 'worldcover_file_description'),
+        'SHORELINE_SOURCE': ('shoreline_shapefile', 'shoreline_shapefile_description'),
+        'REFERENCE_WATER_SOURCE': ('reference_water_file', 'reference_water_file_description')
+    }
+
+    for meta_key, (file_attr, desc_attr) in source_map.items():
+        description = getattr(ancillary_cfg, desc_attr, None)
+        file_path = getattr(ancillary_cfg, file_attr, None)
+        
```

Comment:
```suggestion

```
Having the blank like is okay, but removing the whitespaces here. Same in the below.